File: Python/m00_foundations/ch02_data_input_output/number_doubler.py
def double_number(number):
    """
    Doubles the provided number.

    :param number: The number to be doubled.
    :return: The doubled value of the input number.
    """
    return number * 2


def main():
    print("-" * 21)
    print("Double Number Program")
    print("Type 'exit' to quit the program.")
    print("-" * 21)
    print()

    while True:
        user_input = input("Enter a number: ")

        if user_input.lower() == 'exit':
            print("Exiting program.")
            break

        try:
            input_number = float(user_input)
            doubled_result = double_number(input_number)
            print(f"Doubled number: {input_number} * 2 = {doubled_result:.2f}")
        except ValueError:
            print("Invalid input. Please enter a valid number or type 'exit' to quit the program.")
        print()


if __name__ == "__main__":
    main()

File: JavaScript/m00-foundations/ch02-data-input-output/numberDoubler.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

/**
 * Doubles the provided number.
 *
 * Example:
 * doubleNumber(5); // 10
 *
 * @param {number} number The number to double.
 * @return {number} The doubled value of the input number.
 */
function doubleNumber(number) {
  return number * 2;
}

function main() {
  console.log('---------------------');
  console.log('Double Number Program');
  console.log("Type 'exit' to quit the program.");
  console.log('---------------------');
  console.log();

  const askForNumber = () => {
    rl.question('Enter a number: ', (userInput) => {
      if (userInput.toLowerCase() === 'exit') {
        console.log('Exiting program.');
        rl.close();
        return;
      }

      try {
        const inputNumber = parseFloat(userInput);
        const doubledResult = doubleNumber(inputNumber);
        console.log(`Doubled number: ${inputNumber} * 2 = ${doubledResult.toFixed(2)}`);
      } catch (e) {
        console.log('Invalid input. Please enter a valid number or type \'exit\' to quit the program.');
      }
      console.log();
      askForNumber();
    });
  };

  askForNumber();
}

main();
